### ISSUE TITLE
Test not working?

### ISSUE THREAD

**Issue by ByteFan** [iota-php/client#47]
Hi,
From a fresh install with composer.json
{
    "require": {
        "iota-php/client":"dev-master",
        "php-http/curl-client": "dev-master"

    }
}
on Ubuntu 16.04 I can not run any test? Please what is wrong with my installation?

> php IOTATest.php
PHP Fatal error:  Class 'PHPUnit\Framework\TestCase' not found in /home/iota/php_libs/vendor/iota-php/client/tests/IOTATest.php on line 27

**Comment by robinlehrmann** [comment]
You created a new project, right? (Otherwise, the `iota-php/client` would not be listened as a dependency inside the `composer.json`).
This is ok, but this is not the approach to run the tests of the library.
For this, you have to clone the (this) repository, and run `composer install` inside the cloned directory. Then you can run `./vendor/bin/phpunit`.

In your case, meaning the case of a new project, that uses this library, you have to add a testing framework (like phpunit)  to your (dev-)requirements, and write tests by yourself. (And in this scenario, you would not write tests for the library, as they are already exist).

tldr: You got the wrong scope. You created a new project, but wanted to clone this repository. `composer require` vs. `git clone`.

**Comment by ByteFan** [comment]
Thx. I will look into that!

**Comment by ByteFan** [comment]
I have git cloned the directory then 'php composer.phar install'
 ./vendor/bin/phpunit gives a test failure. It looks some dependencies are not satisfied. Any suggestion ?
PHPUnit 6.5.7 by Sebastian Bergmann and contributors.

...............................................................  63 / 181 ( 34%)
..............................................................S 126 / 181 ( 69%)
SSSSESSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS         181 / 181 (100%)

Time: 29.74 seconds, Memory: 12.00MB

There was 1 error:

1) IOTA\Tests\RemoteApi\AttachToTangleTest::testAction
Http\Discovery\NotFoundException: No message factories found. To use Guzzle, Diactoros or Slim Framework factories install php-http/message and the chosen message implementation.

/home/iota/php_libs/client/vendor/php-http/discovery/src/MessageFactoryDiscovery.php:27
/home/iota/php_libs/client/tests/RemoteApi/AttachToTangleTest.php:42

Caused by
Http\Discovery\Exception\DiscoveryFailedException: Could not find resource using any discovery strategy. Find more information at http://docs.php-http.org/en/latest/discovery.html#common-errors
 - Puli Factory is not available



/home/iota/php_libs/client/vendor/php-http/discovery/src/Exception/DiscoveryFailedException.php:41
/home/iota/php_libs/client/vendor/php-http/discovery/src/ClassDiscovery.php:75
/home/iota/php_libs/client/vendor/php-http/discovery/src/MessageFactoryDiscovery.php:25
/home/iota/php_libs/client/tests/RemoteApi/AttachToTangleTest.php:42

ERRORS!
Tests: 181, Assertions: 633, Errors: 1, Skipped: 55.

**Comment by ByteFan** [comment]
php composer.phar require php-http/guzzle6-adapter fixes

**Comment by ByteFan** [comment]
Now an other error with missing ccurl-cli  :-(

./vendor/bin/phpunit
PHPUnit 6.5.7 by Sebastian Bergmann and contributors.

...............................................................  63 / 181 ( 34%)
..............................................................S 126 / 181 ( 69%)
SSSSESSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS         181 / 181 (100%)

Time: 29.15 seconds, Memory: 12.00MB

There was 1 error:

1) IOTA\Tests\RemoteApi\AttachToTangleTest::testAction
IOTA\Exception: /home/iota/php_libs/client/tests/../ccurl/ccurl-cli not exists

/home/iota/php_libs/client/src/Cryptography/POW/CCurl.php:77
/home/iota/php_libs/client/src/Cryptography/POW/CCurl.php:43
/home/iota/php_libs/client/src/DI/IOTAContainer.php:183
/home/iota/php_libs/client/src/DI/IOTAContainer.php:267
/home/iota/php_libs/client/tests/RemoteApi/AttachToTangleTest.php:55

ERRORS!
Tests: 181, Assertions: 633, Errors: 1, Skipped: 55.

**Comment by robinlehrmann** [comment]
Hi @ByteFan . Thank you. We already have a fix in #46 . The current problem is that we could not versionate the ccurl-cli tool due different architectures of processors. We should create a mock class of POWInterface.

**Comment by ByteFan** [comment]
Yep I understand now. I finaly managed to pass the test. I had not the ccurl binary at the right place. Here is the result:
OK, but incomplete, skipped, or risky tests!
Tests: 181, Assertions: 642, Skipped: 55.

I don't knwow why some tests were automaticaly skipped or "too risky".

**Comment by robinlehrmann** [comment]
Some tests are marked as skipped because several of them are not finished.
